plugin/core/sessions.py

```diff
@@ -220,24 +275,49 @@ def clear_dotted_value(current: dict, dotted: str) -> None:
     current.pop(keys[-1], None)
 
 
-class Session(object):
-    def __init__(self,
-                 config: ClientConfig,
-                 workspace_folders: List[WorkspaceFolder],
-                 client: Client,
-                 on_pre_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_exit: Optional[Callable[[str], None]] = None) -> None:
+class Session(Client):
+
+    @classmethod
+    def name(cls) -> str:
+        raise NotImplementedError()
+
+    @classmethod
+    def needs_update_or_installation(cls) -> bool:
+        return False
+
+    @classmethod
+    def install_or_update(cls) -> None:
+        pass
+
+    @classmethod
+    def standard_configuration(cls) -> ClientConfig:
+        raise NotImplementedError()
+
+    @classmethod
+    def adjust_configuration(cls, configuration: ClientConfig) -> ClientConfig:
+        return configuration
+
+    @classmethod
+    def can_start(cls, window: sublime.Window, initiating_view: sublime.View,
+                  workspace_folders: List[WorkspaceFolder], configuration: ClientConfig) -> Optional[str]:
+        return None
+
+    def on_initialized(self) -> None:
+        pass
+
+    def on_shutdown(self) -> None:
+        pass
+
+    def __init__(self, manager: Manager, settings: Settings, workspace_folders: List[WorkspaceFolder],
+                 config: ClientConfig) -> None:
         self.config = config
+        self.manager = weakref.ref(manager)
+        self.window = manager.window()
         self.state = ClientStates.STARTING
-        self._on_post_initialize = on_post_initialize
-        self._on_post_exit = on_post_exit
         self.capabilities = dict()  # type: Dict[str, Any]
-        self.client = client
         self._workspace_folders = workspace_folders
-        if on_pre_initialize:
-            on_pre_initialize(self)
-        self._initialize()
+        self._progress = {}  # type: Dict[Any, Dict[str, str]]
+        super().__init__(config.name, settings)
```

Comment:
Why is that called? This class doesn't subclass anything (apart from `object`).

Reply:
The base class is `Client`.